airflow/hooks/postgres_hook.py

```diff
@@ -111,3 +125,27 @@ def _serialize_cell(cell, conn):
         :rtype: object
         """
         return cell
+
+    def get_iam_token(self, conn):
+        """
+        Uses AWSHook to retrieve a temporary password to connect to Postgres
+        or Redshift. Port is required. If none is provided, default is used for
+        each service
+        """
+        redshift = conn.extra_dejson.get('redshift')
+        aws_conn_id = conn.extra_dejson.get('aws_conn_id', 'aws_default')
+        aws_hook = AwsHook(aws_conn_id)
+        if conn.port is None:
+            conn.port = 5439 if redshift else 5432
+        if redshift is not None:
+            client = aws_hook.get_client_type('redshift')
+            cluster_creds = client.get_cluster_credentials(DbUser=conn.login,
+                                                           DbName=self.schema or conn.schema,
+                                                           ClusterIdentifier=redshift.get('cluster-identifier'),
+                                                           AutoCreate=False)
+            token = cluster_creds['DbPassword']
+            conn.login = cluster_creds['DbUser']
```

Comment:
I'd be consistent about return vs mutate (and probably _not_ mutate `conn` at all in this function, so make login and port variables that are returned.